src/condor_had/ReplicatorStateMachine.cpp

```diff
@@ -368,26 +365,21 @@ ReplicatorStateMachine::replicaSelectionHandler( Version& newVersion )
 	 * 'replicaSelectionHandler', i.e. selecting the version with greatest
 	 * 'logicalClock' value amongst a group of versions with the same gid
 	 */
-    actualVersionsList.Rewind( );
-    
-    if( actualVersionsList.IsEmpty( ) ) {
+    if( m_versionsList.empty( ) ) {
         return false;
     }
-    Version version;
-    Version bestVersion;
     // taking the first actual version as the best version in the meantime
-    actualVersionsList.Next( bestVersion );
+    Version bestVersion = *m_versionsList.begin();
     dprintf( D_ALWAYS, "ReplicatorStateMachine::replicaSelectionHandler best "
 			"version = %s\n", bestVersion.toString( ).c_str( ) );
-    
-    while( actualVersionsList.Next( version ) ) {
+
+    for (const auto& version : m_versionsList) {
         dprintf( D_ALWAYS, "ReplicatorStateMachine::replicaSelectionHandler "
```

Comment:
This seems like an odd place for a D_ALWAYS.

I like keeping the code as similar as possible, but for new code, maybe calling `std::max_element` might be more clear.